### PR TITLE
run in browser context

### LINKED ISSUE #1: Should use _.runInContext
You should use [_.runInContext](https://lodash.com/docs#runInContext) to avoid potential conflicts with the primary node package.

```js
var _ = require('lodash').runInContext();
```

It can also be used in the browser context:
```js
var _ = root._.runInContext();
```

### PR DESCRIPTION
**Pull request by yoshokatana**
fixes #1

### CODE REVIEW

**Comment by jdalton** [comment]
I think it should be:
```js
root._ = factory(root._.runInContext());
```

This will overwrite the existing one but devs can always use `noConflict()` to get the previous version.

**Comment by TakenPilot** [comment]
Is the purpose of assigning directly to `root._` from `root._.runInContext()` in the browser to avoid other mixins?  If we use more than one mixin (like lodash-deep), wouldn't they disappear?

**Comment by jdalton** [comment]
lodash-deep will just extend whatever lodash is on `root._`. By using you pave the existing but allow devs to restore with `_.noConflict`. If you don't want to go that route, which is ok, there isn't really a reason to use `_.runInContext` when exporting for the browser.

**Comment by TakenPilot** [comment]
Hmm.

This browser code is all-around a weird thing the more I look at it.  We're mixing in these functions on [line 99](https://github.com/nymag/lodash-ny-util/blob/run-in-context/index.js#L99), which makes AMD and node work.

For the browser, we're passing in whatever `window._` is at the moment and then [line 99](https://github.com/nymag/lodash-ny-util/blob/run-in-context/index.js#L99) is adding those functions to it.  But then on [line 20](https://github.com/nymag/lodash-ny-util/blob/run-in-context/index.js#L20) we're using mixin again.

I think it's a good thing to runInContext, but we should be consistent about it.  We're not doing it for AMD, but we are for node, and then we aren't for the browser.  We should pick one.

I'm going to test each of these to see what code actually plays well with other mixins in each of these scenarios.

**Comment by TakenPilot** [comment]
After playing around in the browser, I'm okay with all three versions.  : )

Considering how small this package is currently (until we pull more generic logic out of our internal repositories to share, as more of our projects become public), lets go with assigning directly to `root._`.  It'll allow us to be more specific.  If we run into issues as this repo grows, it might be better to deviate from lodash-deep and be more specific with the ordering for how we mix into lodash functionality.

:+1:

**Comment by yoshokatana** [comment]
:+1:

**Comment by jdalton** [comment]
:+1:

File: lodash-ny-util.js [hunk 1]
```diff
@@ -4,24 +4,37 @@
...
+  /**
+   *
+   * @param {string|function} fn
+   * @returns {function(this:null)}
+   */
+  function convertToPromisePartial(fn) {
+    fn = _.isFunction(fn) ? fn : _[fn];
+    var result = _.partial.bind(null, fn, _);
+    if (arguments.length > 1) {
+      result = result.apply(null, Array.prototype.slice.call(arguments, 1));
```
Comment (jdalton) [inline lodash-ny-util.js]: `Array.prototype.slice` -> `_.slice(arguments, 1)`

Comment (TakenPilot) [inline lodash-ny-util.js]: Ah, nice. I'll add that in a quick PR tomorrow.

**Pull request merged**